pkg/resourceinterpreter/customized/webhook/serviceresolvers_test.go

```diff
@@ -0,0 +1,139 @@
+package webhook
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func Test_resolveCluster(t *testing.T) {
+	type args struct {
+		svc  *corev1.Service
+		port int32
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name: "cluster ip without 443 port",
+			args: args{
+				svc: &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "one", Name: "alfa"},
+					Spec: corev1.ServiceSpec{
+						Type:      corev1.ServiceTypeClusterIP,
+						ClusterIP: "hit",
+						Ports: []corev1.ServicePort{
+							{Port: 1234, TargetPort: intstr.FromInt32(1234)},
+						}}},
+				port: 443,
+			},
+			wantErr: assert.Error,
+		},
+		{
+			name: "cluster ip",
+			args: args{
+				svc: &corev1.Service{
+					ObjectMeta: metav1.ObjectMeta{Namespace: "one", Name: "alfa"},
+					Spec: corev1.ServiceSpec{
+						Type:      corev1.ServiceTypeClusterIP,
+						ClusterIP: "hit",
```

Comment:
```suggestion
						ClusterIP: "10.10.10.10",
```
Why not use a normal IP?